forge-game/src/main/java/forge/game/spellability/SpellAbility.java

```diff
@@ -716,6 +712,9 @@ public final void addOptionalCost(OptionalCost cost) {
         // Thus, to protect the original's set from changes, we make a copy right here.
         optionalCosts = EnumSet.copyOf(optionalCosts);
         optionalCosts.add(cost);
+        if (!cost.getPip().equals("")) {
```

Comment:
this might be problematic when the raise/reduce costs are done in the wrong order

Reply:
I think this would also be solved by that other issue?

Reply:
I just made an additional issue